configserver/src/main/java/com/yahoo/vespa/config/server/tenant/ActivateLock.java

```diff
@@ -34,7 +34,7 @@ public synchronized boolean acquire(TimeoutBudget timeoutBudget, boolean ignoreL
         }
     }
 
-    public synchronized void release() {
+    public void release() {
         if (curatorLock.hasLock()) {
             curatorLock.unlock();
```

Comment:
Won't this become racy now and possibly result in multiple unlocks of the same lock?

Reply:
The lock is taken and released in one method, if the lock is not acquired an exception is thrown, but finally is run anyway (trying to release the lock). If the lock was not acquired, you will get an exception stating that you do not own the lock (the locking implementation in InterProcessMutex keeps track of which thread owns the lock). Maybe it's better to only call release() when you have acquired the lock. I'll look into it.